torch/nn/functional.py

```diff
@@ -565,3 +565,31 @@ def pad(input, pad, mode='constant', value=0):
             return _functions.thnn.ReplicationPad3d(*pad)(input)
     else:
         raise NotImplementedError("Only 4D and 5D padding is supported for now")
+
+
+# distance
+
+def pairwise_distance(input, p=2):
+    r"""
+    Computes the batchwise pairwise distance between vectors v1,v2:
+
+        .. math ::
+            \Vert x \Vert _p := \left( \sum_{i=1}^n  \vert x_i \vert ^ p \right) ^ {1/p}
+
+        Args:
+            x (Tensor): input tensor containing the two input batches
+            p (real): the norm degree. Default: 2
+
+        Shape:
+            - Input: :math:`(2, N, D)` where `D = vector dimension`
```

Comment:
Actually there's no point in requiring a single input to the module and the function. You could easily compute outputs on two batches with different models, and ignore the cat before feeding them into the pairwise distance. Can you change the signature to `pairwise_distance(x1, x2, p=2, eps=1e-6)` (and do the same with the module)? It's also helpful to be able to change the eps.